### ISSUE TITLE
finetuning the model

### ISSUE THREAD

**Issue by soans1994** [ZheC/Realtime_Multi-Person_Pose_Estimation#244]
Dear Author,

I tried finetuning the model from the pretarined pose_iter_440000.caffemodel with the custom set of images and annotation. But the loss is very high and the infernece detection is bad. I used the follwing bash file

**Comment by cohogain** [comment]
HI @soans1994, did you resolve this issue? I am experiencing similar situation. Thanks

**Comment by soans1994** [comment]
> HI @soans1994, did you resolve this issue? I am experiencing similar situation. Thanks

Yes, I just

**Comment by soans1994** [comment]
> HI @soans1994, did you resolve this issue? I am experiencing similar situation. Thanks

Try running the all steps and train command again. Seems like some bug. Sometimes when trying the train command, the loss is empty. If you continue the training, the model output is bad.

**Comment by cohogain** [comment]
Thanks for your reply, I will try this. May I ask roughly how large was your dataset you used to fine-tune the model?

**Comment by soans1994** [comment]
I tried less up to 200 images and model gives good results. Loss drops quickly until 50000 iteration and then slow drop of loss for further iterations

**Comment by cohogain** [comment]
Ok great thank you. I am finetuning with 150 images but the model seems to loose all accuracy and gives many incorrect predictions or even none. The loss also does not drop further than 100. I am not sure if I keep training or if I have preconfig incorrect.

**Comment by soans1994** [comment]
Yes. Check your annotation files too. Json file should be similar to coco format, and number of key points. I didn’t do any changes.

**Comment by cohogain** [comment]
This also might be it, I am currently using 12 keypoints (eyes, shoulders, elbow, wrist, hips, knees). I will annotate the rest. Can I ask one final question, when annotating images, do you initialise keypoints you cannot see (covered ear / leg) or do you leave unannotated? Currently I am not initialising them but I see in coco-annotator tool you can set as NOT-VISIBLE. That is my last question, thanks for all your help!

**Comment by soans1994** [comment]
I will check my annoted  file and attach it tomorrow.

**Comment by cohogain** [comment]
Thanks you, I appreciate that!

**Comment by soans1994** [comment]
I set the invisible points to zero. But i think your method is better. It also depends on your application. I use infrared images, so sometimes i annotate the keypoints even if it is not visible since the infared images are very low quality and we cannot see human parts, to get better detection of keypoints.

[sleep2.txt](https://github.com/ZheC/Realtime_Multi-Person_Pose_Estimation/files/6128508/sleep2.txt)

**Comment by cohogain** [comment]
Ok interesting, my data is in similar format so. I believe uninitialised keypoints are given value (1000,1000) by the program. Would you be willing to also attach the dataset of ~200 images you used to train your model with? It would be very helpful as if I cannot replicate your results an the same dataset, I can know that the issue involves my training application. Is this something you can still access? I appreciate your help so far

**Comment by soans1994** [comment]
The dataset i used is related to project my professor is working for a private company. So i cannot share it. But, i believe any image works. The image is 80x60 original dimension from small IR camera, but i resized to 400x300 for easy annotation and testing. try for a small dataset even 50 images works.
![16](https://user-images.githubusercontent.com/30766648/110928555-5cb3ee80-836a-11eb-858b-960885526e6f.png)

**Comment by cohogain** [comment]
OK thats no problem I will try some testing based on your info, thanks for helping me.
